@kiva/kv-components/vue/stories/KvCarousel.stories.js

```diff
@@ -0,0 +1,210 @@
+import KvCarousel from '../KvCarousel.vue';
+import KvCarouselSlide from '../KvCarouselSlide.vue';
+import KvLoadingSpinner from '../KvLoadingSpinner.vue';
+import KvButton from '../KvButton.vue';
+
+const randomHexColor = () => Math.floor(Math.random() * 16777215).toString(16);
+
+// This is not an actual loan card template, just a vague
+// approximation to make testing in the stories nicer
+const generateLoanCardTemplate = () => {
+	const cardCopy = [
+		'A loan of $5,450 helps a member to buy flour, eggs, lard, sugar, sweets and other...',
+		'A loan of $1,125 helps to face the financial problem of covering tuition fees.',
+		'A loan of $450 helps to purchase store goods for resale.',
+	];
+
+	return `
+		<div style="width: 336px">
+			<img src="https://via.placeholder.com/336x252/${randomHexColor()}/000000" class="tw-w-full tw-rounded tw-mb-2">
+			<h3>Card Title</h3>
+			<h4 class="tw-my-1">$${Math.floor(Math.random() * (10000 - 50) + 50)} to go</h4>
+			<p class="tw-mt-1 tw-mb-9">${cardCopy[Math.floor(Math.random() * cardCopy.length)]}</p>
+			<kv-button
+				variant="primary"
+			>
+				Read more
+			</kv-button>
+		</div>`;
+};
+
+const defaultCarouselSlides = `
+	<template #slide1>
+		<img src="https://via.placeholder.com/300x220/${randomHexColor()}/000000" class="tw-w-full">
+	</template>
+	<template #slide2>
+		<img src="https://via.placeholder.com/300x220/${randomHexColor()}/000000" class="tw-w-full">
+	</template>
+	<template #slide3>
+		<img src="https://via.placeholder.com/300x220/${randomHexColor()}/000000" class="tw-w-full">
+	</template>
+	<template #slide4>
+		<img src="https://via.placeholder.com/300x220/${randomHexColor()}/000000" class="tw-w-full">
+	</template>
+`;
+
+export default {
+	title: 'KvCarousel',
+	component: KvCarousel,
+};
+
+export const Default = () => ({
+	components: {
+		KvCarousel,
+		KvCarouselSlide,
+	},
+	template: `
+		<kv-carousel style="max-width: 400px;" :number-of-slides="4">
+			${defaultCarouselSlides}
+		</kv-carousel>
+	`,
+});
+
+export const LoopFalse = () => ({
+	components: {
+		KvCarousel,
+		KvCarouselSlide,
+	},
+	template: `
+		<kv-carousel style="max-width: 400px;" :embla-options="{ loop: false }" :number-of-slides="4">
+			${defaultCarouselSlides}
+		</kv-carousel>
+	`,
+});
+
+export const MultipleLoanCards = () => ({
+	components: {
+		KvCarousel,
+		KvCarouselSlide,
+		KvButton,
+	},
+	template: `
+		<kv-carousel
+			:embla-options="{ loop: false }"
+			:number-of-slides="5"
```

Comment:
Rather than having to pass the number of slides and use dynamically generated named slots, have you looked into computing some of this by reaching into the data from `this.$slots` inside KvCarousel.vue? Not sure how far it could get you but could lead to a simpler component API.

Reply:
I simplified this a little bit, it still requires slot names, but they can be anything. also removed the 'number of slides' prop